### ISSUE TITLE
orig_callback conflicts with faye-websocket

### ISSUE THREAD

**Issue by ShogunPanda** [alebsack/rack-fiber_pool#2]
In mperham/rack-fiber_pool#24 you changed the `async.callback` to `async.orig_callback`.

As you can see here https://github.com/faye/faye-websocket-ruby/blob/master/lib/faye/websocket.rb#L72, that made this gem not working.

Why did you make that change? Can we put it under a configuration variable so is not universal?

Thanks and regards.
  Shogun

**Comment by alebsack** [comment]
The change is there in order to 'hide' the asynchronous behavior from downstream rack middlewares and applications.  For example, Sinatra apps that use the async.callback mechanism to signal Thin that it is streaming data would call the callback proc, skipping over rack-fiber_pool's handler.

The change simply ensures that the request is always handled with synchronous behavior within the fiber.

Can you provide a code example where this breaks?

**Comment by alebsack** [comment]
I've looked a little more into Faye-websocket, and I fail to see how the rack-fiber_pool middleware would even be beneficial. 

Exposing the async.callback to it basically overrides the features of the pool, except for wrapping the request in a fiber. But Faye-websocket doesn't require a fiber context in order to run. By exposing it, the callback would be run twice: once by Faye-websocket and again by rack-fiber_pool once the websocket connection closes. The webserver (in most cases, thin) does not know how to handle two calls to the callback.

**Comment by ShogunPanda** [comment]
My use case was a Rails 4 app, with websocket-rails (and therefore faye-websocket) under thin and rack-fiber_pool.

Basically, in https://github.com/faye/faye-websocket-ruby/blob/master/lib/faye/websocket.rb#L72, faye-websocket looks for `async.callback` to send data back to the client using HTTP 101.

As rack-fiberpool is renaming the callback, the data was never sent and the client was waiting indefinitely for a response.

In other words, your behavior is correct when everything is managed inside the pool, but there are side case where the original behavior might be needed. 
Maybe you're right and that override should be on by default, but at that time you could take my pull request and reverse the option so that someone can explicitly disable this behavior if needed.

**Comment by ShogunPanda** [comment]
Closed for inactivity. Probably useless since ActionCable is now out.
